dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -164,15 +164,21 @@ public DoraLoadJob(
         path, Preconditions.checkNotNull(Scheduler.getInstance()).getActiveWorkers().size());
 
     try {
-      mUfsStatusIterator = mUfs.listStatusIterable(
-          ufsSyncRootUri.toString(), ListOptions.defaults().setRecursive(true), null, 0);
-      if (mUfsStatusIterator == null) {
-        mUfsStatusIterator = Collections.emptyIterator();
+      UfsStatus rootUfsStatus = mUfs.getStatus(ufsSyncRootUri.toString());
+      if (rootUfsStatus != null && rootUfsStatus.isFile()) {
+        rootUfsStatus.setUfsFullPath(ufsSyncRootUri);
+        mUfsStatusIterator = Iterators.singletonIterator(rootUfsStatus);
       } else {
-        mUfsStatusIterator = Iterators.transform(mUfsStatusIterator, (it) -> {
-          it.setUfsFullPath(ufsSyncRootUri.join(it.getName()));
-          return it;
-        });
+        mUfsStatusIterator = mUfs.listStatusIterable(
```

Comment:
i would prefer to listStatusIterable first and if it's null then we try to getStatus like what we did for other places. We usually try to load directories so we can save one API call in more common case

Reply:
1. The cost of adding a single getStatus() call to this job is ignorable. We are doing many listing, work rpcs & actual data loadings anyways.
2. The behavior of listStatus is not quite defined. I'm afraid that in some UFS implementations, listing a file still gives you the ufs status of the file while others give you an empty list. If the former one is the case, listing before getStatus will give you the wrong result.